SmartDeviceLink/public/SDLAlertView.m

```diff
@@ -0,0 +1,146 @@
+//
+//  SDLAlertView.m
+//  SmartDeviceLink
+//
+//  Created by Nicole on 11/10/20.
+//  Copyright © 2020 smartdevicelink. All rights reserved.
+//
+
+#import "SDLAlertView.h"
+
+#import "SDLAlertAudioData.h"
+#import "SDLArtwork.h"
+#import "SDLError.h"
+#import "SDLSoftButtonObject.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface SDLAlertView()
+
+@property (copy, nonatomic, nullable) SDLAlertCanceledHandler canceledHandler;
+
+@end
+
+@implementation SDLAlertView
+
+static const float TimeoutDefault = 0.0;
+static const float TimeoutMinCap = 3.0;
+static const float TimeoutMaxCap = 10.0;
+static NSTimeInterval _defaultTimeout = 5.0;
+
+- (instancetype)init {
+    self = [super init];
+    if (!self) { return nil; }
+
+    _timeout = TimeoutDefault;
+
+    return self;
+}
+
+- (instancetype)initWithText:(NSString *)text buttons:(NSArray<SDLSoftButtonObject *> *)softButtons {
+    self = [self init];
+    if (!self) { return nil; }
+
+    _text = text;
+    self.softButtons = softButtons;
+
+    return self;
+}
+
+- (instancetype)initWithText:(nullable NSString *)text secondaryText:(nullable NSString *)secondaryText tertiaryText:(nullable NSString *)tertiaryText timeout:(nullable NSNumber<SDLFloat> *)timeout showWaitIndicator:(nullable NSNumber<SDLBool> *)showWaitIndicator audioIndication:(nullable SDLAlertAudioData *)audio buttons:(nullable NSArray<SDLSoftButtonObject *> *)softButtons icon:(nullable SDLArtwork *)icon {
+    self = [self initWithText:text buttons:softButtons];
+    if (!self) { return nil; }
+
+    _secondaryText = secondaryText;
+    _tertiaryText = tertiaryText;
+    self.timeout = timeout.doubleValue;
+    _showWaitIndicator = showWaitIndicator.boolValue;
+    self.audio = audio;
```

Comment:
I had to use `self` in order to get a copy of the object used to set the value, which I assume is the intended behavior. Let me know if not.